File: lang/nodejs/src/fun/vac/util/RandomFactory.js
"use strict"

exports.launch = () => {
    // preserve for consistent interface
}

exports.integerN = (min, max) => {
    return min + Math.floor(Math.random() * (max - min))
}

exports.generateInteger = () => {
    return this.integerN(0, 2147483647)
}

exports.generateEven = () => {
    return this.generateInteger() >>> 1 << 1
}

exports.generateOdd = () => {
    return this.generateInteger() | 1
}

File: lang/python/src/fun/vac/util/RandomFactory.py
import random

def seed() -> None:
    random.seed()

def integerN(min: int, max: int) -> int:
    return random.randint(min, max)

def generateInteger() -> int:
    return integerN(0, 2_147_483_647)

def generateEven() -> int:
    return generateInteger() >> 1 << 1

def generateOdd() -> int:
    return generateInteger() | 1
